component/lambda/functions/billing-set-prices.py

```diff
@@ -1,140 +1,139 @@
 from datetime import datetime, timedelta
-from typing import Iterator, Literal, NotRequired, Optional, TypedDict, cast
-import boto3
+import logging
+from typing import Iterator, cast
 
-from si_logger import logger
-from si_redshift import Redshift
+from si_lambda import SiLambda
 from si_lago_api import (
-    LagoApi,
     ExternalSubscriptionId,
     LagoHTTPError,
-    LagoInvoicesResponse,
     LagoSubscription,
 )
 from si_types import OwnerPk
 
 
-session = boto3.Session()
-redshift = Redshift.from_env(session)
-lago = LagoApi.from_env(session)
-
 paul = OwnerPk("01GW0KXH4YJBWC7BTBAZ6ZR7EA")
 plan_code = "launch_pay_as_you_go"
 billable_metric_code = "resource-hours"
 cost_per_resource_hour = 0.007
 
 
-def get_invoice_month():
-    """
-    Get the start of the month for the next invoice we will send out.
-
-    Prices will be set based on the user's usage during this month.
-    """
-    now = datetime.now()
-    day_in_invoice_month = now
-    if now.day > 5:
-        day_in_invoice_month = day_in_invoice_month - timedelta(days=6)
-    return day_in_invoice_month.replace(
-        day=1, hour=0, minute=0, second=0, microsecond=0
-    )
+class BillingSetPrices(SiLambda):
+    def get_invoice_month(self):
+        """
+        Get the start of the month for the next invoice we will send out.
 
-
-def get_subscription(
-    external_subscription_id: ExternalSubscriptionId,
-) -> LagoSubscription:
-    try:
-        return cast(
-            LagoSubscription,
-            lago.get(f"/api/v1/subscriptions/{external_subscription_id}").json()[
-                "subscription"
-            ],
+        Prices will be set based on the user's usage during this month.
```

Comment:
query: Can you elaborate slightly? I don't quite follow why if the day is >5 that you would want to run this logic

Reply:
Will do! The reason is that we need to use *last* month's price if we're past the 1st of the month but last month's invoices aren't finalized yet (due to the grace period and the fact that last month's invoice will pick up any price overrides automatically until it is finalized).